services/garage-backend/src/controllers/thingController.ts

```diff
@@ -14,11 +14,16 @@ import { serializeThing } from '../serializers/things';
 // POST /things
 export const createThing = async (req: Request, res: Response, next: NextFunction) => {
   try {
+    const thingDocument = await Thing.findOne({ name: req.body.name });
+    if (thingDocument) {
+      next(new BadRequestError('Thing with same name exists in the Database'));
```

Comment:
```suggestion
      next(new BadRequestError('Thing with same name exists in the Database'));
      return;
```
don't forget to return